lib/sycamore/sycamore/query/client.py

```diff
@@ -0,0 +1,255 @@
+#!/usr/bin/env python
+
+# This is a simple CLI for Sycamore Query that lets you run queries against an index,
+# and see the generated query plan and result.
+#
+# Example:
+#   poetry run sycamore/query/client.py --index const_ntsb \
+#       "How many incidents were there in Washington in 2023?"
+#
+# Use --help for more options.
+
+import argparse
+import json
+import logging
+from typing import Optional, Tuple
+import os
+import uuid
+
+import matplotlib.pyplot as plt
+from opensearchpy import OpenSearch
+from opensearchpy.client.indices import IndicesClient
+import structlog
+
+import sycamore
+from sycamore.llms.openai import OpenAI, OpenAIModels
+from sycamore.transforms.query import OpenSearchQueryExecutor
+from sycamore.utils.cache import S3Cache
+
+
+from sycamore.query.execution.sycamore_executor import SycamoreExecutor
+from sycamore.query.logical_plan import LogicalPlan
+from sycamore.query.planner import LlmPlanner
+from sycamore.query.schema import OpenSearchSchema
+from sycamore.query.visualize import visualize_plan
+
+from rich.console import Console
+
+
+console = Console()
+
+
+DEFAULT_OS_CONFIG = {"search_pipeline": "hybrid_pipeline"}
+DEFAULT_OS_CLIENT_ARGS = {
+    "hosts": [{"host": "localhost", "port": 9200}],
+    "http_compress": True,
+    "http_auth": ("admin", "admin"),
+    "use_ssl": True,
+    "verify_certs": False,
+    "ssl_assert_hostname": False,
+    "ssl_show_warn": False,
+    "timeout": 120,
+}
+
+
+def configure_logging(logfile: Optional[str] = None, log_level=logging.WARN):
+    """Configure logging for LUnA query execution."""
+
+    class CustomLoggerFactory(structlog.stdlib.LoggerFactory):
+        """Custom logger factory that directs output to a file."""
+
+        def __init__(self, file_handler):
+            super().__init__()
+            self.file_handler = file_handler
+
+        def __call__(self, *args, **kwargs):
+            logger = super().__call__(*args, **kwargs)
+            logger.addHandler(self.file_handler)
+            return logger
+
+    logging.basicConfig(level=log_level)
+
+    if logfile:
+        processors = [
+            structlog.contextvars.merge_contextvars,
+            structlog.processors.TimeStamper(fmt="iso"),
+            structlog.stdlib.add_logger_name,
+            structlog.stdlib.add_log_level,
+            structlog.stdlib.PositionalArgumentsFormatter(),
+            structlog.processors.StackInfoRenderer(),
+            structlog.processors.format_exc_info,
+            structlog.processors.UnicodeDecoder(),
+            structlog.processors.JSONRenderer(),
+        ]
+
+        file_handler = logging.FileHandler(logfile)
+        file_handler.setLevel(log_level)
+        structlog.configure_once(
+            processors=processors,
+            logger_factory=CustomLoggerFactory(file_handler),
+            wrapper_class=structlog.make_filtering_bound_logger(log_level),
+            cache_logger_on_first_use=True,
+        )
+
+
+class LunaClient:
```

Comment:
I know we want to get this in quickly, so I'm okay if this is in a separate PR, but presumably we should rename "Luna" to "SycamoreQuery" or something similar?

Reply:
Yes, that's the plan. I'll probably do a quick follow up PR.